core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -584,6 +614,14 @@ private[spark] class TaskSchedulerImpl(
     executorIdToTaskCount.getOrElse(execId, -1) > 0
   }
 
+  /**
+   * Get a snapshot of the currently blacklisted nodes for the entire application.  This is
+   * thread-safe -- it can be called without a lock on the TaskScheduler.
+   */
+  def nodeBlacklist(): scala.collection.immutable.Set[String] = {
```

Comment:
things would still compile & work as expected if I only changed the type here.  But I actually thought it was useful for the type to indicate that this was returning an immutable set, to reinforce the thread-safety implied by immutable collections.  Yes it means the user of this needs to think about about why this one is immutable, but that seems like a good thing to me.

(I don't feel strongly about this though)